### PR TITLE
JanusGraphSerializerBaseIT: run against latest code

### PR DESCRIPTION
**Pull request by li-boxuan**
Currently, JanusGraphSerializerBaseIT runs against janusgraph/janusgraph:latest image on Docker Hub. This means the integration test runs unreleased (latest) driver code against released server code. Even worse, this integration test is not run by the CI due to a misconfiguration in janusgraph-driver module (see #3445).

This commit moves the integration test from janusgraph-driver to janusgraph-server module and refactors the test suite, so that we can run latest unreleased driver code against latest unreleased server code.

See https://github.com/janusgraph/janusgraph/commit/b96aa6d26f74d3dc8f7404212bddeefe2d0790b4#r94972076

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### CODE REVIEW

**Comment by li-boxuan** [comment]
Integration test is picked up by CI: https://github.com/JanusGraph/janusgraph/actions/runs/3836785957/jobs/6532369099#step:6:305

File: janusgraph-server/src/test/resources/empty-sample.groovy [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import org.apache.tinkerpop.gremlin.server.util.LifeCycleHook
```
Comment (farodin91) [inline janusgraph-server/src/test/resources/empty-sample.groovy:1]: Is this file needed?

File: janusgraph-server/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphSerializerBaseIT.java [hunk 2]
```diff
@@ -17,26 +17,34 @@
...
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestInfo;
-import org.testcontainers.junit.jupiter.Container;
-import org.testcontainers.junit.jupiter.Testcontainers;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 
-@Testcontainers
 public abstract class JanusGraphSerializerBaseIT {
```
Comment (farodin91) [inline janusgraph-server/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphSerializerBaseIT.java:32]: Great idea to move to the server. I would like to add combat test in the future, but they are probably better suited for the janusgraph-dist.

**Review by farodin91 — APPROVED** [review]
LGTM! @li-boxuan  Thank you.

File: janusgraph-server/src/test/resources/empty-sample.groovy (hunk 1 above)
Comment (li-boxuan) [reply janusgraph-server/src/test/resources/empty-sample.groovy:1]: Yes, otherwise I got `The traversal source [g] for alias [g] is not configured on the server` error. Not sure if there's a better way to solve this.

Comment (li-boxuan) [reply janusgraph-server/src/test/resources/empty-sample.groovy:1]: Will merge this for now. If anyone has a better idea on handling the error, let me know or create a PR.

**Comment by li-boxuan** [comment]
Backporting seems not trivial. Since this is just a refactor of some test cases, I'll abandon the backport.
